core/src/main/java/hudson/model/WorkspaceListener.java

```diff
@@ -3,25 +3,77 @@
 import hudson.ExtensionList;
 import hudson.ExtensionPoint;
 import hudson.FilePath;
+import hudson.Util;
+import hudson.console.ConsoleLogFilter;
+
+import java.io.OutputStream;
 
 public abstract class WorkspaceListener implements ExtensionPoint {
     
     /**
      * Called after a workspace is deleted successfully.
      * @param project
+     * @deprecated as of TODO_BEFORE_MERGE Use {@link #afterDelete(Job)}.
```

Comment:
The annotation is stored within the byte-code, the javadoc was not.  So when compiling against something with the binary only (ie no access to the source code) static analysis/the compiler can know that the method is deprecated.

That's the theory anyway...  What sucks is that you need both and the annotation doesn't take a String with a reason!
Also see the Note on [this document](http://docs.oracle.com/javase/1.5.0/docs/guide/javadoc/deprecation/deprecation.html#how).  the Oracle compiler can and may choose to ignore the javadoc tag in future versions.